src/swig/CMakeLists.txt

```diff
@@ -71,3 +73,42 @@ if (GUILE_FOUND)
   set (GUILE_EXTENSION_PATH ${_REL_GUILE_EXTENSION_PATH})
   install (TARGETS nlopt_guile LIBRARY DESTINATION ${GUILE_EXTENSION_PATH})
 endif ()
+
+
+if (JNI_FOUND AND Java_FOUND AND SWIG_FOUND)
```

Comment:
build fails if guile is enabled because of this line, maybe it should be moved to the guile target only:
set_source_files_properties (nlopt.i PROPERTIES SWIG_FLAGS "-scmstub"):
```
swig error : Unrecognized option -scmstub
```

Reply:
The Guile binding probably needs to do the same as what Java does for the `-package nlopt` flag, i.e., setting the global `SWIG_FLAGS` property before `swig_add_library` and unsetting it afterwards instead of using `set_source_files_properties`. I can make that change and update this pull request with it. The problem with using `set_source_files_properties` is that the same source `nlopt.i` is used for all target languages.

Reply:
This should be fixed in the new revision (commit 32a4a64e7b087fb0cec719ccd794ed5a8fe2f585).